### PR TITLE
Add second output to trigger node

### PR DESCRIPTION
**Pull request by dceejay**
This PR add the option of making the second output of the trigger node as a separate output - the msg remains the same in either case - but it may make the wiring clearer for some users.

And adds tests

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Proposed changes

As per above

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/node-red/node-red/blob/master/CONTRIBUTING.md)
- [x] For non-bugfix PRs, I have discussed this change on the mailing list/slack team.
- [x] I have run `grunt` to verify the unit tests pass
- [x] I have added suitable unit tests to cover the new/changed functionality

### CODE REVIEW

File: packages/node_modules/@node-red/nodes/core/function/89-trigger.html [hunk 1]
```diff
@@ -74,16 +81,19 @@
...
         color:"#E6E0F8",
         defaults: {
+            name: {value:""},
             op1: {value:"1", validate: RED.validators.typedInput("op1type")},
             op2: {value:"0", validate: RED.validators.typedInput("op2type")},
             op1type: {value:"val"},
             op2type: {value:"val"},
             duration: {value:"250",required:true,validate:RED.validators.number()},
             extend: {value:"false"},
+            second: {value:false},
```
Comment (knolleary) [inline packages/node_modules/@node-red/nodes/core/function/89-trigger.html]: I don't think we need both `second` and `outputs` as new properties on the node. Just setting `outputs=2` ought to be enough for both the editor and runtime to know what they need to do.

Comment (dceejay) [reply packages/node_modules/@node-red/nodes/core/function/89-trigger.html]: I thought outputs has to be numeric type and a checkbox has to be boolean ? so having two is not a great hardship vs mangling the type on prepare and on save ? Only one is used on the js side

Comment (knolleary) [reply packages/node_modules/@node-red/nodes/core/function/89-trigger.html]: Having two inflates the flow file size for absolutely every user of the trigger node, all for the sake of not having to add one line of JavaScript in `oneditprepare`.

Comment (dceejay) [reply packages/node_modules/@node-red/nodes/core/function/89-trigger.html]: OK - fixed.

**Comment by jeancarl** [comment]
Should this option be visible only when the mode is `wait for`, since there is no "second message" in the other two modes (`wait to be reset` and `resend it every`)?

**Comment by dceejay** [comment]
sensible thought...

**Pull request merged**